Utils/Types/AtomicQueue.hpp

```diff
@@ -0,0 +1,90 @@
+// ======================================================================
+// \title  AtomicQueue.hpp
+// \brief  Lightweight lock-free non-allocating single reader single writer queue
+// ======================================================================
+
+#ifndef UTILS_TYPES_ATOMIC_QUEUE_HPP
+#define UTILS_TYPES_ATOMIC_QUEUE_HPP
+
+#include <atomic>
+
+namespace Types {
+
+// Note: PlatformSizeType is probably generally larger than we need,
+// but it should still be an efficient size to manipulate,
+// and it's guaranteed to be unsigned, which is crucial.
+template<class E, PlatformSizeType SIZE>
```

Comment:
Typically I expect `FwSizeType` to be used in contexts such as this.  `PlatformSizeType` is really a default set by the platform developer that flows into the `FwSizeType` if the project chooses not to override the default.